grakn-graph/src/main/java/ai/grakn/graph/internal/RoleTypeImpl.java

```diff
@@ -124,6 +154,10 @@ public void innerDelete(){
             throw new ConceptException(ErrorMessage.CANNOT_DELETE.getMessage(getName()));
         } else {
             super.innerDelete();
+
+            //Clear all internal caching
+            cachedRelationTypes.clear();
+            cachedDirectPlayedByTypes.clear();
```

Comment:
Nope. This check: `boolean hasHasRoles = getVertex().edges(Direction.IN, Schema.EdgeLabel.HAS_ROLE.getLabel()).hasNext();` before hand means that you won't have any relation types to notify.